authentication-pre-request.js

```diff
@@ -0,0 +1,40 @@
+const { Header } = require('postman-collection');
+
+const { collectionVariables } = pm;
+
+const oauth2Request = {
+    url: 'https://api.oregonstate.edu/oauth2/token',
+    method: 'POST',
+    header: 'Content-Type:application/x-www-form-urlencoded',
+    body:`grant_type=client_credentials&scope=
+      &client_id=${collectionVariables.get('clientId')}
+      &client_secret=${collectionVariables.get('clientSecret')}`
+};
+
+const bearerToken = collectionVariables.get('bearerToken');
+const tokenExpirationDate = collectionVariables.get('tokenExpirationDate');
+if (pm.environment.name === 'LOCAL') {
+    const token = btoa(
+      `${collectionVariables.get('username')}:${collectionVariables.get('password')}`);
```

Comment:
If you use 2 spaces for indentation, this line can be put in one line.
Also, just my two cents, I will load all variables at the beginning and it will probably make the rest of codes cleaner like this:

```js
const { collectionVariables, request: { headers } } = pm;
const bearerToken = collectionVariables.get('bearerToken');
const tokenExpirationDate = collectionVariables.get('tokenExpirationDate');
const username = collectionVariables.get('username');
const password = collectionVariables.get('password');
const environmentName = pm.environment.name;  // this can't be destructured since `environment` is reserved variable in POSTMAN already.
```